libsql/internal/http/http.go

```diff
@@ -159,3 +159,33 @@ func generatePostBody(sql string, sqlParams params) (*postBody, error) {
 
 	return &postBody, nil
 }
+
+// httpResultsAlternative is an alternative struct for unmarshalling the response
+// see more info here: https://github.com/libsql/sqld/issues/466
+type httpResultsAlternative struct {
+	Results *resultSet `json:"results"`
+	Error   string     `json:"error"`
+}
+
+func unmarshalResponse(body []byte, result *[]httpResults) error {
+	err := json.Unmarshal(body, result)
+	if err == nil {
+		return nil
+	}
+
+	var alternativeResults []httpResultsAlternative
+	errArray := json.Unmarshal(body, &alternativeResults)
+	if errArray != nil {
+		return err
+	}
+
+	convertedResult := make([]httpResults, len(alternativeResults))
+	for _, alternativeResult := range alternativeResults {
+		convertedResult = append(convertedResult, httpResults{
+			Results: alternativeResult.Results,
+			Error:   &httpErrObject{Message: alternativeResult.Error}})
+	}
+	result = &convertedResult
```

Comment:
@luisfvieirasilva @athoscouto I must misunderstand something here. Unless I'm mistaken, this statement has no effect and results in `result` being empty when the alternative response format is being used.

Didn't you mean to do `*result = convertedResult`?

See the difference in simplified example below:
```
package main

import "fmt"

func test(a *[]int) {
	b := []int{1, 2, 3}
	a = &b
}

func main() {
	c := []int{4, 5}
	fmt.Println(c)
	test(&c)
	fmt.Println(c)
}

prints:
[4 5]
[4 5]
```
while:
```
package main

import "fmt"

func test(a *[]int) {
	b := []int{1, 2, 3}
	*a = b
}

func main() {
	c := []int{4, 5}
	fmt.Println(c)
	test(&c)
	fmt.Println(c)
}

prints:
[4 5]
[1 2 3]
```

is current behaviour what you expected or did you mean `*result = convertedResult`?